src/draive/conversation/call.py

```diff
@@ -7,17 +10,54 @@
 ]
 
 
+@overload
+async def conversation_completion(
+    *,
+    instruction: str,
+    input: ConversationMessage | StringConvertible,  # noqa: A002
+    memory: Memory[ConversationMessage] | None = None,
+    toolset: Toolset | None = None,
+    stream: Literal[True] = True,
+) -> AsyncIterator[ConversationMessage]:
+    ...
+
+
+@overload
 async def conversation_completion(
     *,
     instruction: str,
     input: ConversationMessage | StringConvertible,  # noqa: A002
     memory: Memory[ConversationMessage] | None = None,
     toolset: Toolset | None = None,
+    stream: Literal[False] = False,
 ) -> ConversationMessage:
+    ...
+
+
+async def conversation_completion(
+    *,
+    instruction: str,
+    input: ConversationMessage | StringConvertible,  # noqa: A002
+    memory: Memory[ConversationMessage] | None = None,
+    toolset: Toolset | None = None,
+    stream: bool = False,
+) -> AsyncIterator[ConversationMessage] | ConversationMessage:
     conversation: Conversation = ctx.state(Conversation)
-    return await conversation.completion(
-        instruction=instruction,
-        input=input,
-        memory=memory or conversation.memory,
-        toolset=toolset or conversation.toolset,
-    )
+
+    if stream:
```

Comment:
This `if` statement looks weird.

```py
return await conversation.completion(
            instruction=instruction,
            input=input,
            memory=memory or conversation.memory,
            toolset=toolset or conversation.toolset,
            stream=stream,
        )
```

Reply:
yes it is, I have added a comment - pyright does not handle types correctly if not using literal True/False when calling this